### ISSUE TITLE
KDE Plasma: Keyboard does not work with Wayland

### ISSUE THREAD

**Issue by klemensn** [NixOS/nixpkgs#223074]
### Describe the bug
I use a ThinkPad X250 with a broken internal keyboard via external USB keyboard.
Using Plasma 5 with Wayland instead of X11 results in loss of keyboard input.
Replugging the USB device does not help.
It does work with Wayland in other desktop environments, so I assume this is a bug in KDE's Plasma.

### Steps To Reproduce
1. use
```
xserver = {
  displayManager = {
    defaultSession = "plasmawayland";
    sddm.enable = true;
  };
  desktopManager.plasma5.enable = true;
};
```
2. login
3. try to type

### Expected behavior
Keyboard input ought to work.

### Additional context
Keyboard works with
- Plasma/X11
- GNOME/Wayland
- Pantheon (not sure what's used there)

Pretty sure this used to work one or two releases ago, since I had to actively switch to X11 to unbreak my system.
Same breakages with unstable.

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.104, NixOS, 22.11 (Raccoon), 22.11.3333.83607dae4e0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(kn): `"nixos"`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Hexadecimaaal** [comment]
I think this also reproduces in X11 for me? I am upgrading flake from 5f4e07deb7c44f27d498f8df9c5f34750acf52d2 to 4bb072f0a8b267613c127684e099a70e1f6ff106, and keyboard breaks after login into kde from sddm. I suspect it's not from higher layers since kdeconnect remote input still works somehow. I guess it might be from https://github.com/NixOS/nixpkgs/commit/77e09950a6095ca89aad69778ee65d9d5b4db3f2 ? CC @K900

**Comment by K900** [comment]
That would be a weird regression. Does it affect all windows or just Plasma? The only possibly relevant issue I've seen upstream is https://bugs.kde.org/show_bug.cgi?id=467519

**Comment by Hexadecimaaal** [comment]
Seems be all windows. Tried Firefox and Telegram and neither works. Konsole and systemsettings5 definitely doesn't. I do have fcitx5 installed and `XMODIFIERS` is set to `@im=fcitx`, and it does still work with kdeconnect remote input.

**Comment by K900** [comment]
What if you turn it off?

**Comment by Hexadecimaaal** [comment]
Nope, still not working.

**Comment by Hexadecimaaal** [comment]
Additional details: here I'm upgrading from plasma 5.27.0 to 5.27.3, kde frameworks 5.103.0 to 5.104.0, qt 5.15.8 to the same version. Reproducible on both kernel 5.15.104 and 6.1.21.

**Comment by K900** [comment]
And this is on X11, correct? Do the events show up in `xev` at least?

**Comment by K900** [comment]
Then we're probably looking at an xinput issue. How about if you run `libinput debug-events`?

**Comment by Hexadecimaaal** [comment]
Alright, it just struck me and I stopped my xkeysnail mapping service and the keyboard works now. Sorry for the confusion. My config reads:

<details>
<summary>Details</summary>

```nix
{
  environment.systemPackages = [ pkgs.xkeysnail ];

  systemd.user.services."xkeysnail" = {
    description = "xkeysnail daemon";
    wantedBy = [ "graphical-session.target" ];
    partOf = [ "graphical-session.target" ];
    serviceConfig = {
      ExecStartPre= "-${pkgs.xorg.xhost}/bin/xhost +SI:localuser:root";
      ExecStart = "${pkgs.xkeysnail}/bin/xkeysnail --quiet --watch ${./config.py}";
      RestartSec = "5";
      Restart = "always";
    };
  };
}
```
</details>

and this seems to be broken silently after the upgrade. This is probably unrelated to the original issue.

**Comment by klemensn** [comment]
> That would be a weird regression. Does it affect all windows or just Plasma?

Plasma's menu/application search, terminal emulator, browser, nothing works.

**Comment by klemensn** [comment]
> This is probably unrelated to the original issue.

No `xkeysnail` used here.

**Comment by K900** [comment]
> > That would be a weird regression. Does it affect all windows or just Plasma?
> 
> Plasma's menu/application search, terminal emulator, browser, nothing works.

How about other windows, i.e. not-Plasma ones?

**Comment by klemensn** [comment]
> How about other windows, i.e. not-Plasma ones?

How do I get "not-Plasma" windows in the Plasma DE?  What does that mean?
I've tried all desktop applications I have, none of them receive keyboard input.

**Comment by K900** [comment]
OK, and this is specifically on Wayland, correct? Does X11 work? Are events displayed in the Wayland session if you run `libinput debug-events`?

**Comment by klemensn** [comment]
> OK, and this is specifically on Wayland, correct? Does X11 work?

Yes, only Wayland is broken. X11 works.
I also flipped `xserver.libinput.enable = true;` (configuration.nix(5) does not tell what this does) to `false`, rebuilt without upgrading and rebooted, but with no avail -- same results.

>  Are events displayed in the Wayland session if you run `libinput debug-events`?

I have yet to establish other means of keyboard input to be able to type/run anything under Wayland.

**Comment by K900** [comment]
Does the mouse work?

**Comment by klemensn** [comment]
> Does the mouse work?

Yes, internal touchpad and trackpoint as well as an external bluetooth mouse all work under both Wayland and X11.

**Comment by K900** [comment]
Try logging in from a TTY, running `export QT_LOGGING_RULES="*.debug=true"`, then `startplasma-wayland &> plasma.log`. Then post the contents of `wayland.log`

**Comment by klemensn** [comment]
Same results with KDE Connect's remote input.
Mouse and keyboard work in X11, only mouse works in Wayland.

The android app even tells me that `keyboard input is not supported by the paired device` when under Wayland.

**Comment by K900** [comment]
That's... very interesting. At this point I'd probably try reporting this upstream, though finding the right regression window would be nice too.

**Comment by klemensn** [comment]
> Try logging in from a TTY, running `export QT_LOGGING_RULES="*.debug=true"`, then `startplasma-wayland &> plasma.log`. Then post the contents of `wayland.log`

Did that, clicked on the menu, typed into the search bar without seeing input, then logged out via grahical menu and mouse.
https://gist.github.com/klemensn/9c411ed07c3dd562f1182834f447e46c

**Comment by K900** [comment]
Hmm, I wonder why that's not picking up any of the kwin logs then.

**Comment by ArmirDev** [comment]
Has anyone found a solution to it? Currently facing it too...

**Comment by DanielJoyce** [comment]
Same here. External keyboard plugged into laptop doesn't work. Even when I close the lid so I use an external monitor
